File: mycuration/src/main/java/com/phicdy/mycuration/view/FeedSearchView.java
package com.phicdy.mycuration.view;

import android.support.annotation.NonNull;

public interface FeedSearchView {
    void startFeedUrlHookActivity(@NonNull String url);
    void showProgressDialog();
    void dismissProgressDialog();
    void load(@NonNull String url);
    void showInvalidUrlErrorToast();
    void showGenericErrorToast();
    void showAddFeedSuccessToast();
    void finishView();
    void setSearchViewTextFrom(@NonNull String url);
}

File: feature_feed_search/src/main/java/com/phicdy/mycuration/feedsearch/FeedSearchView.kt
package com.phicdy.mycuration.feedsearch

interface FeedSearchView {
    fun startFeedUrlHookActivity(url: String)
    fun showProgressBar()
    fun dismissProgressBar()
    fun load(url: String)
    fun showInvalidUrlErrorToast()
    fun showGenericErrorToast()
    fun showAddFeedSuccessToast()
    fun finishView()
    fun setSearchViewTextFrom(url: String)
    fun trackFailedUrl(url: String)
}
